auth/src/Login.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+
+class Login extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      username: "",
+      password: ""
+    };
+  }
+
+  input = e => {
+    e.preventDefault();
+    console.log("%c input has been triggered", "font-size:18; color:red;");
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  render() {
+    return (
+      <form
+        onSubmit={e => {
+          e.preventDefault();
+          this.props.login({
+            username: this.state.username,
```

Comment:
note here that the submission is the entire state object so you could use  onSubmit={this.props.login(this.state) }